packages/gatsby/src/commands/build-html.ts

```diff
@@ -314,12 +315,22 @@ export const doBuildPages = async (
   try {
     await renderHTMLQueue(workerPool, activity, rendererPath, pagePaths, stage)
   } catch (error) {
+    const pageData = await getPageData(error.context.path)
+
+    // These objs can be quite large, so we want to limit the character count
+    const pageDataMessage = `Data for failed page "${
+      error.context.path
+    }": ${JSON.stringify(pageData, null, 2).slice(0, 50000)}`
+
     const prettyError = await createErrorFromString(
       error.stack,
       `${rendererPath}.map`
     )
+
     const buildError = new BuildHTMLError(prettyError)
```

Comment:
```suggestion
    const buildError = new BuildHTMLError({ prettyError, pageDataMessage })
```

Let's only print one error for the failure vs. two

Reply:
I'm not sure I see a way to do this gracefully without a bunch of changes. The `BuildHTMLError` `class` needs to be altered to accept another param, and then we need to add the param to a property of the error object, and then add it to each place where the error is caught and then logged. Am I correct with this assumption? Is that what you had in mind?

Reply:
This is the only place this Error is used.

Reply:
Yeah, this is the only place the error is used, but when the error is thrown and then caught that's when it logs all the info, and this error is caught in various places. I think we have to add additional context in each of the locations that it's thrown to log out an extra property from this error object.

There's no way that I can see, with the current properties that this Error object has, to add the log to it. Each of the Error's properties are styled in a way when it's logged out that it wouldn't make sense to append the `pageData` to any of those strings.

Reply:
oh right sorry — I didn't look at this closely and had in mind our `reporter` which does take structured data.